assets/js/utils/style-attributes-utils.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable @wordpress/no-unsafe-wp-apis */
+/**
+ * External dependencies
+ */
+import { __experimentalUseColorProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { isFeaturePluginBuild } from '../settings/blocks/feature-flags';
+import { isString, isObject } from '../types/type-guards';
+
+type WithClass = {
+	className: string;
+};
+
+type WithStyle = {
+	style: Record< string, unknown >;
+};
+
+const parseStyle = ( style: unknown ): Record< string, unknown > => {
+	if ( isString( style ) ) {
+		return JSON.parse( style ) || {};
+	}
+
+	if ( isObject( style ) ) {
+		return style;
+	}
+
+	return {};
+};
+
+const parseSpacingStyle = (
+	spacing: Record< string, unknown >
+): Record< string, unknown > => {
+	const keys = [ 'margin' ];
+
+	const getValueOrDefault = ( value: unknown ) => {
+		return isString( value ) && value.length > 0 ? value : '0';
+	};
```

Comment:
I don't think we should return `0` as the default value here, it should be `undefined` instead. The element may have spacing rules defined in CSS, using `0` will override its styling.